### ISSUE TITLE
Add support for a 'disabled' state

### ISSUE THREAD

**Issue by bbriggs** [rnelson0/puppet-local_user#17]
Would like to have a user exist but disable login (perhaps expire the password?) when `state => 'disabled'`

**Comment by bbriggs** [comment]
For linux users I was thinking of setting the shell to `/bin/false` and the login date to something far in the past. Will that be sufficient?

**Comment by rnelson0** [comment]
I think it's reasonable to lock the account and set the shell to false unless a shell is specified. The login date should not be changed. This way files could still be created in the user's homedir or with their ownership, but remote access and local logins are disabled.

Is this something you would be interested in working on?

**Comment by bbriggs** [comment]
Yeah, we have a need for it where I'm using it. So just set the shell to false then? That will still allow [S]FTP logins.

**Comment by rnelson0** [comment]
Unless the $shell param is set, yes. If someone wants to override it with
eksh or something else, they can, but a simple `local_user{'wiki': ensure
=> disabled}` will have sane settings.
